shared/nm-utils/nm-test-utils.h

```diff
@@ -1089,6 +1091,8 @@ __define_nmtst_static(02, 1024)
 __define_nmtst_static(03, 1024)
 #undef __define_nmtst_static
 
+#if defined (__NM_UTILS_H__) || defined (NM_UTILS_H)
```

Comment:
`shared/nm-utils/nm-test-utils.h` does this at several places already:

```

#ifdef NM_SETTING_IP_CONFIG_H
static inline void
...
#endif /* NM_SETTING_IP_CONFIG_H */
```
and
```
#if defined(__NM_SIMPLE_CONNECTION_H__)
»···»···type_g = nm_setting_lookup_type (type);
#else
»···»···type_g = nm_connection_lookup_setting_type (type);
#endif
```

The reason is, that `nm-test-utils.h` is a header-only "library" of testing utilities. It is used by all our tests. The right way to use it is:
  - including "nm-test-utils.h" as last file. 
  - before, including other headers that turn on functionality in "nm-test-utils.h".

The alternative, would either require multiple header files, and you explicitly include them. Or even not using a header-only approach. IMO both makes it only more cumbersome to write tests. As it is, it's convenient to use. Also, almost all our tests do either include "libnm", "libnm-utils", or "libnm-core" (thus, usually, the code is available). This test is an exception, but other parts of `nm-test-utils.h` are still interesting to have here.

It's not documented, but it's actually impossible to get wrong, because you either lack functionality (and don't care), or your code won't compile.

This also means our include guards are meaningful and should have well-defined names. `#pragma once` would not work here. But I don't mind.